commons/src/lib.rs

```diff
@@ -91,16 +91,34 @@ pub fn ensure_query_params(
     Ok(())
 }
 
-/// Make sure client requested a valid content type.
-pub fn ensure_content_type(
+/// Make sure the client can accept the provided media type.
+pub fn validate_content_type(
     headers: &HeaderMap,
     content_type: &'static str,
 ) -> Result<(), GraphError> {
-    let content_json = header::HeaderValue::from_static(content_type);
+    if !headers.contains_key(header::ACCEPT) {
+        return Ok(());
+    }
+
+    let full_type = header::HeaderValue::from_static(content_type);
+    let wildcard = header::HeaderValue::from_static("*");
+    let double_wildcard = header::HeaderValue::from_static("*/*");
+    let top_type = content_type.split("/").next().unwrap_or("");
+    let top_type_wildcard = header::HeaderValue::from_str(&format!("{}/*", top_type));
+    assert!(
+        top_type_wildcard.is_ok(),
+        format!("could not form top-type wildcard from {}", top_type)
+    );
 
     if !headers
```

Comment:
Build a vector of acceptable content types 

```suggestion
    let acceptable_content_types: Vec<&str> =
        vec![full_type, wildcard, double_wildcard, top_type_wildcard];
    let header_value = headers.get(header::ACCEPT);

	if acceptable_content_types.iter().any(|c| c == header_value) {
		Ok(())
	} else {
		Err(GraphError::InvalidContentType)
	}
```

Reply:
done in ade9cb68 -> 73e8e06a